torch/distributed/fsdp/flat_param.py

```diff
@@ -432,6 +433,65 @@ def _get_shard_metadata(
                 shard_param_indices[-1] - shard_param_indices[0] + 1
         return tuple(shard_param_offsets), shard_param_indices
 
+    @staticmethod
+    def _get_shard_unpadded(
+        tensor: Tensor,
+        rank: int,
+        world_size: int,
+    ) -> Tuple[Tensor, int]:
```

Comment:
Should we consider to avoid using `Tuple` in the future but use `NamedTuple` for better readability during this code refactor?

Reply:
Makes sense. I will do this in follow-ups.